src/core/Domain.ml

```diff
@@ -159,12 +158,7 @@ struct
         (Pp.pp_sep_list (fun fmt (lbl, tp) -> Format.fprintf fmt "%a : %a" Ident.pp_user lbl pp_con tp)) fields
     | Prf ->
       Format.fprintf fmt "*"
-    | Cof (Cof.Join phis) ->
-      Format.fprintf fmt "join[%a]" (Pp.pp_sep_list pp_con) phis
-    | Cof (Cof.Meet phis) ->
-      Format.fprintf fmt "meet[%a]" (Pp.pp_sep_list pp_con) phis
-    | Cof (Cof.Eq (r, s)) ->
-      Format.fprintf fmt "eq[%a,%a]" pp_con r pp_con s
+    | Cof phi -> Cof.dump_cof_f pp_con pp_con fmt phi
```

Comment:
Dump is for debugging or whatever, but this seems to be a real pretty printer, at least in name. Should `pp_con` etc. be renamed to `dump_con`?

Reply:
Good point. I don't know. Maybe another solution is to rename `Cof.dump` to `Cof.pp`? :laughing: The pretty printer is not printing out anything beautiful here---perhaps that's your point?

Reply:
So one thing is that often `pp` requires access to the proof state, whereas `dump` is meant to be callable by anyone anywhere. I don't have an opinion on what we call these specific cases though, except to say: every construct should in principle have both a `dump` and a `pp`. If the same code is used for both, then we should do something like define `pp = dump` or whatever.